pkg/skaffold/update/update.go

```diff
@@ -31,20 +32,32 @@ import (
 // Fot testing
 var (
 	GetLatestAndCurrentVersion = getLatestAndCurrentVersion
+	readConfig                 = config.ReadConfigFileNoCache
+	getEnv                     = os.Getenv
 )
 
 const latestVersionURL = "https://storage.googleapis.com/skaffold/releases/latest/VERSION"
 
 // IsUpdateCheckEnabled returns whether or not the update check is enabled
 // It is true by default, but setting it to any other value than true will disable the check
-func IsUpdateCheckEnabled() bool {
+func IsUpdateCheckEnabled(configfile string) bool {
 	// Don't perform a version check on dirty trees
 	if version.Get().GitTreeState == "dirty" {
 		return false
 	}
 
-	v := os.Getenv(constants.UpdateCheckEnvironmentVariable)
-	return v == "" || strings.ToLower(v) == "true"
+	return isUpdateCheckEnabledByEnvOrConfig(configfile)
+}
+
+func isUpdateCheckEnabledByEnvOrConfig(configfile string) bool {
+	if v := getEnv(constants.UpdateCheckEnvironmentVariable); v != "" {
+		return strings.ToLower(v) == "true"
+	}
+	cfg, err := readConfig(configfile)
```

Comment:
the config file reading logic should be in the config package with the other config parameters.

Reply:
ah ok. i will move reading "UpdateCheck" to config.